src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -199,11 +206,103 @@ public override void OnEventWritten(string name, object? payload)
                     finally
                     {
                         activity.Stop();
+                        this.RecordDuration(activity, payload, hasError: true);
                     }
                 }
 
                 break;
         }
     }
+
+    private void RecordDuration(Activity? activity, object? payload, bool hasError = false)
+    {
+        if (SqlActivitySourceHelper.DbClientOperationDuration.Enabled == false)
+        {
+            return;
+        }
+
+        var tags = new TagList
+        {
+            { SemanticConventions.AttributeDbSystem, SqlActivitySourceHelper.MicrosoftSqlServerDatabaseSystemName },
+        };
+
+        if (activity != null && activity.IsAllDataRequested)
+        {
+            if (hasError)
+            {
+                var exceptionType = activity.Tags.SingleOrDefault(t => t.Key == SemanticConventions.AttributeExceptionType);
+
+                if (exceptionType.Value != null)
+                {
+                    tags.Add(SemanticConventions.AttributeErrorType, exceptionType.Value);
+                }
+            }
+
+            var serverAddress = activity.GetTagItem(SemanticConventions.AttributeServerAddress);
+
+            if (serverAddress != null)
+            {
+                tags.Add(SemanticConventions.AttributeServerAddress, serverAddress);
+            }
+
+            var port = activity.GetTagItem(SemanticConventions.AttributeServerPort);
+
+            if (port != null)
+            {
+                tags.Add(SemanticConventions.AttributeServerPort, port);
+            }
+        }
+        else
+        {
+            _ = this.commandFetcher.TryFetch(payload, out var command);
+
+            if (command != null)
+            {
+                if (hasError)
+                {
+                    if (this.exceptionFetcher.TryFetch(payload, out Exception? exception) && exception != null)
+                    {
+                        tags.Add(SemanticConventions.AttributeErrorType, exception.Message);
```

Comment:
the message has high cardinality,  this should be the full name of the exception type.
We also added `db.response.status_code` recently - ptal at some details on how to capture it 
https://github.com/open-telemetry/semantic-conventions/blob/v1.28.0/docs/database/sql.md